packages/@okta/vuepress-theme-prose/mixins/AnchorHistory.vue

```diff
@@ -26,9 +26,19 @@ export default {
       }
 
       this.paddedHeaderHeight = this.getPaddedHeaderHeight()
-
+      
       const anchorOffsets = this.anchors.map(
-        anchor => anchor.parentElement.offsetTop
+        /* 
+          In case of a few pages like error codes (/docs/reference/error-codes/), we are using our custom template
+          for showing header-anchor (ErrorCodes.vue) where the generated HTML has different template than the default
+          vuepress template for showing anchor tags.
+          The anchor tag is not a direct child of h2/h3/h4 in this case but instead, is the grandchild of these header elements.
+          Hence, we need to add a separate check for these routes.
+          Refer - https://oktainc.atlassian.net/browse/OKTA-483028
+        */
+        anchor => this.$page.path.includes('docs/reference/error-codes') &&
+        anchor.parentElement.tagName.toLowerCase() !== 'h4' ? anchor.parentElement.parentElement.offsetTop :
+        anchor.parentElement.offsetTop
```

Comment:
That looks too specific for one page we need a more generic solution.

If you remove `HEADER_TO_CONTENT_GAP` https://github.com/okta/okta-developer-docs/blob/a17175d8228d6135549faf662f9bb3830ed7a4f7/packages/%40okta/vuepress-theme-prose/mixins/AnchorHistory.vue#L16 will it also fix the bug? If so, I would prefer that.

We should sync with dev docs team in their @developer-docs-team channel. I'd love to have navigation implemented similar to Netlify's documentation https://docs.netlify.com/site-deploys/manage-deploys/ that would simply implementation

Reply:
The issue has nothing to do with `HEADER_TO_CONTENT_GAP`. As mentioned in the code comment, it happens because we have written general logic of calculating `anchorOffsets` which works for all cases but error-codes since error codes has its own HTML template for anchor tags and does not use the default vuepress anchor tags template. Hence, this solution was aimed at just this page.

Implementing a new navigation is a big task IMO and should not block the fix for this issue. It can be taken up as a separate task.